src/beamlime/resources/images/generators.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from functools import partial
+
+import numpy as np
+
+
+def adds_up_to(number: float, frags: list = None, max_split: int = 5) -> np.ndarray:
+    if number == 0 or len(frags) == max_split - 1:
+        return np.array(frags + [number])
+    new_frag = np.random.random() * number
+    return adds_up_to(number - new_frag, frags + [new_frag])
+
+
+def split_number(
+    intensity: float, min_intensity: float, max_frame: int = 128, max_split: int = 5
+) -> np.array:
+    if intensity < min_intensity:
+        return np.zeros(max_frame)
+
+    margin = max(0.1, intensity - (max_split * min_intensity))
+    fragments = adds_up_to(margin, frags=[], max_split=max_split)
+    distributed_margin = fragments + min_intensity
+    distributed_intensity = np.zeros(max_frame)
+    distributed_intensity[: len(distributed_margin)] = distributed_margin
+    np.random.shuffle(distributed_intensity)
+    return distributed_intensity
+
+
+def add_noise(frame: np.ndarray, noise_range=0.2) -> np.ndarray:
+    """
+    Add uniform random noise within the range [0, noise_range) on the image.
+    """
+    return frame + np.random.uniform(0, abs(noise_range), frame.shape)
+
+
+def fake_2d_detector_img_generator(
```

Comment:
Any reason for not returning `sc.Variable`?

Reply:
Because it is only used to generate `raw-data`.
The fake data feeder that I added in #30 will wrap this into a dictionary.

So the `offline.data_feeder` will generate what `kafka-listener` will eventually return for testing.